BlobForwarder/index.js

```diff
@@ -42,29 +42,45 @@ module.exports = async function main(context, myBlob) {
     context.log.warn('logs format is invalid');
     return;
   }
-  let compressedPayload;
-  let payloads = generatePayloads(buffer, context);
-  for (const payload of payloads) {
-    try {
-      compressedPayload = await compressData(JSON.stringify(payload));
-      try {
-        await retryMax(httpSend, NR_MAX_RETRIES, NR_RETRY_INTERVAL, [
+
+  let logLines = appendMetaDataToAllLogLines(buffer, context);
+  await compressAndSend(logLines, context);
+};
+
+function compressAndSend(data, context) {
+  return compressData(JSON.stringify(getPayload(data, context)))
+    .then((compressedPayload) => {
+      if (JSON.stringify(compressedPayload).length > NR_MAX_PAYLOAD_SIZE) {
```

Comment:
Hey @Sivakumar3695 , I believe this is incorrect. The previous code was doing the following:
```
JSON.stringify(payload).length + JSON.stringify(log).length <
      NR_MAX_PAYLOAD_SIZE
```

But please note that this code was **incorrect**, as it was checking that the size of the payload plus the size of the log to be appended to the payload didn't exceed the `NR_MAX_PAYLOAD_SIZE` _before being compressed_. Instead, what we need to do is to check that the size of the resulting **compressed** payload is less than `NR_MAX_PAYLOAD_SIZE`.

So, at this point of your new code, `compressedPayload` already represents a compressed payload body to be sent to the Logs API (it no longer represents a JSON). So, we shouldn't use `JSON.stringify()` here, we just need to check that the size of `compressedPayload` is less than `NR_MAX_PAYLOAD_SIZE`.

Reply:
Thanks for pointing this out. I was not aware of compressedPayload size calculations! 
Great! 🙂